docs/DeveloperGuide.md

```diff
@@ -311,36 +310,81 @@ case associated with the name Thomas, who is of the age 13 and has a postal code
+Step 4. `FindCommandParser#parse()` is called. The `Age` `13`, `Name` `Thomas` and `SubPostal` (a substring of a
+postal) `612` are validated then extracted, and a `FindPredicate` is constructed which is the predicate used to test
+whether the person in the Dengue Case List matches all the given arguments.
 
-Step 5. `FindCommand#execute()` will get the most updated list of filtered cases based on the values given from
-`ArgumentMultimap#getValue()` and shows it on the User Interface.
+Step 5. `FindPredicate` is passed back into `FindCommandParser#parse()` and a new `FindCommand` is constructed, taking
+in this `FindPredicate`.
+
+Step 6. `FindCommand#execute()` will get the most updated list of filtered cases based on the `FindPredicate#test()`
+and displays it on the User Interface along with a success message.
 
 The following sequence diagram summarises what happens when a user executes a Prefix find operation:
 
 ![PrefixFindSequenceDiagram](images/PrefixFindSequenceDiagram.png)
 
-The following activity diagram summarises what happens when a user executes a Prefix find operation:
-
-![PrefixFindActivityDiagram](images/PrefixFindActivityDiagram.png)
-
 #### Design Considerations
 
-**Aspect: How Prefix find handle input that does not make sense (Numerics in names, non-existent postal codes
-or age past 200)**
+**Aspect: How Prefix find handle input that does not make sense (Numerics in names, non-existent postal codes)**
 
-* **Alternative 1 (current choice):** Accepts the user input and executes the find command as per usual.
+* **Alternative 1:** Accepts the user input and executes the find command as per usual.
     * Pros: Allows for user freedom in cases that there may be people with Numerics in names, and the underlying code
       would not have to change if new postal codes were to be implemented.
     * Cons: In the case of the input being erroneous, there is no indication that the for the user that it may be due to
       what they keyed in.
 
-* **Alternative 2:** Displays a message indicating that the input may be erroneous for each of the available prefix.
-    * Pros: In the case of the input being erroneous, there would be an indication that the for the user that their input
-      may be unintended.
+* **Alternative 2 (current choice):** Displays a message indicating that the input is erroneous for the first erroneous
+  prefix detected.
+    * Pros: In the case of the input being erroneous, there would be an indication that the for the user that their
+      input may be unintended.
     * Cons: Less flexibility and requires changes to the code base if new postal codes are added.
 
+### Find-by-range feature
+
+#### Implementation
+
+This feature is largely similar to the find-by-prefix feature, except that the user can input up to two dates and up to 
+two ages, a start date `sd/` and an end date `ed/` instead of a specific date `d/` or a start age `sa/` and an end age
```

Comment:
To split the clauses better.

```suggestion
two ages, a start date `sd/` and an end date `ed/` instead of a specific date `d/`, or a start age `sa/` and an end age
```